src/coreclr/jit/emitxarch.cpp

```diff
@@ -1480,9 +1481,20 @@ emitter::code_t emitter::AddEvexPrefix(const instrDesc* id, code_t code, emitAtt
     {
         code |= BBIT_IN_BYTE_EVEX_PREFIX;
 
+        // enable ymm embedded rounding
+        if (emitComp->compOpportunisticallyDependsOn(InstructionSet_AVX10v2))
+        {
+            code &= ~(uBIT_IN_BYTE_EVEX_PREFIX);
+        }
         if (!id->idHasMem())
         {
-            // embedded rounding case.
+            // ymm embedded rounding case.
+            if (attr == EA_32BYTE)
+            {
+                assert(emitComp->compIsaSupportedDebugOnly(InstructionSet_AVX10v2));
+                code &= ~(uBIT_IN_BYTE_EVEX_PREFIX);
+            }
```

Comment:
As per the above, shouldn't this only be getting stripped in the case that we're known to be setting the rounding mode?

Reply:
Actually, nevermind on this one. We've already checked that `b` is set and we don't have a memory operand; meaning we are doing a rounding mode.